src/app.js

```diff
@@ -6,8 +26,356 @@ import _ from 'underscore';
+      //NEED TO FIND A WAY TO
+      $('#status-messages').removeClass('success')
+      $('#status-messages').addClass('failure')
+      $('#status-messages').append(`Failed to save trip`)
+      console.log(response);
+      tripList.remove(model)
+
+      handleValidationFailures(response.responseJSON["errors"]);
+    },
+  });
+
+};
+
+const readFormData = function readFormData() {
+  const tripData = {};
+
+  TRIP_FIELDS.forEach((field) => {
+    // select the input corresponding to the field we want
+    let inputElement = $(`#add-trip-form input[name="${ field }"]`);
+
+    if(field == 'continent') {
+      inputElement = $(`#add-trip-form select[name="${ field }"]`);
+      console.log(inputElement);
+      console.log(inputElement);
+      const string = inputElement.val();
+      const value = string.charAt(0).toUpperCase() + string.slice(1)
+      console.log(value);
+      if (value != '') {
+        tripData[field] = value;
+      }
+      inputElement.val('');
+    }
+    else if(field == 'weeks') {
+      const value = parseInt(inputElement.val());
+      console.log(`week: ${value} ${typeof value}`);
+      if (value != '') {
+        tripData[field] = value;
+      }
+      inputElement.val('');
+    }
+    else if(field == 'cost') {
+      const value = parseInt(inputElement.val());
+      if (value != '') {
+        tripData[field] = value;
+        console.log(tripData[field]);
+      }
+      inputElement.val('');
+    } else {
+      const value = inputElement.val();
+      if (value != '') {
+        tripData[field] = value;
+      }
+      inputElement.val('');
+    }
+  });
+
+  return tripData;
+};
+
+const handleValidationFailures = function handleValidationFailures(errors) {
+  // Since these errors come from a Rails server, the strucutre of our
+  // error handling looks very similar to what we did in Rails.
+  console.log('inhandlevalidationfaiilurs');
+  for (let field in errors) {
+    for (let problem of errors[field]) {
+      reportStatus('error', `${field}: ${problem}`);
+    }
+  }
+};
+
+const reportStatus = function reportStatus(status, message) {
+  console.log(`Reporting ${ status } status: ${ message }`);
+
+  // Should probably use an Underscore template here.
+  const statusHTML = `<p class="${ status }">${ message }</p>`;
+
+  // note the symetry with clearStatus()
+  $('#status-messages').append(statusHTML);
+  $('#status-messages').removeClass('success')
+
+  $('#status-messages').addClass('failure')
+  $('#status-messages').show();
+};
+
+
+const clearStatus = function clearStatus() {
+  $('#status-messages').html('');
+  $('#status-messages').hide();
+};
+
+const readFilterForm = function readFilterForm() {
+
+}
+
+
+$(document).ready(() => {
+  $('main').addClass('hide')
+  tripTemplate = _.template($('#trip-template').html());
+  aboutTemplate = _.template($('#about-template').html());
+  let filters = {}
+
+  $('#show-trips').on('click', function() {
```

Comment:
As a note, it's theoretically possible that someone could click on this button to to show the trips before fetch completes.  Instead I would use a Backbone Event handler and show the trips after it completes.